Code/Engine/GameEngine/AI/SensorComponent.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <Core/Utils/IntervalScheduler.h>
+#include <Core/World/World.h>
+#include <GameEngine/GameEngineDLL.h>
+
+class ezPhysicsWorldModuleInterface;
+
+struct EZ_GAMEENGINE_DLL ezMsgSensorVisibleObjectsChanged : public ezEventMessage
+{
+  EZ_DECLARE_MESSAGE_TYPE(ezMsgSensorVisibleObjectsChanged, ezEventMessage);
+
+  ezArrayPtr<ezGameObjectHandle> m_VisibleObjects;
+};
+
+//////////////////////////////////////////////////////////////////////////
+
+class EZ_GAMEENGINE_DLL ezSensorComponent : public ezComponent
```

Comment:
I know we often don't document components much, but it would still be nice to actually have some basic description here, what this component does and how to use it.